packages/components/src/flyout/flyout/component.js

```diff
@@ -21,8 +21,8 @@ import { useFlyout } from './hook';
 
 /**
  *
- * @param {import('../../ui/context').PolymorphicComponentProps<import('../types').Props, 'div'>} props
```

Comment:
Just to clarify, why is `Flyout` itself non-polymorphic?

Reply:
It's because TypeScript would otherwise complain when expanding `{ ...otherProps }` into `<FlyoutComponent />`.

Since the `as` prop from `Flyout` can't be applied to `FlyoutContent` (since `FlyoutContent` is non-polymorphic), I thought that `Flyout` itself should also be marked as non-polymorphic.

Do you have an alternative approach to this?

Reply:
Oh I see, the as prop would be applied to FlyoutContent, not some wrapper in Flyout itself? This makes sense then.